ksqldb-engine/src/main/java/io/confluent/ksql/logicalplanner/nodes/SelectNode.java

```diff
@@ -31,15 +39,41 @@ public SelectNode(
     super(input);
     Objects.requireNonNull(selectClause, "selectClause");
 
-    // expression after check only works with '*''
-    if (selectClause.getSelectItems().size() > 1
-        || !(selectClause.getSelectItems().get(0) instanceof AllColumns)) {
-      throw new UnsupportedOperationException("Only `SELECT *` supported");
-    }
+    final ImmutableList<LogicalColumn> inputSchema = input.getOutputSchema();
+    final Builder<LogicalColumn> outputSchemaBuilder = ImmutableList.builder();
 
-    outputSchema = selectClause.getSelectItems().stream().flatMap(
-        s -> input.getOutputSchema().stream()
-    ).collect(ImmutableList.toImmutableList());
+    selectClause.getSelectItems().forEach(
+        s -> {
+          if (s instanceof AllColumns) {
+            outputSchemaBuilder.addAll(inputSchema);
+          } else {
+            final Expression e = ((SingleColumn) s).getExpression();
+            // to-do: maybe extend using
+            // `UnqualifiedColumnReferenceExp` and `QualifiedColumnReferenceExp`
+            if (e instanceof ColumnReferenceExp) {
+              final ColumnName columnName = ((ColumnReferenceExp) e).getColumnName();
+              final ImmutableList<LogicalColumn> matchingInputColumns = inputSchema.stream()
+                  .filter(logicalColumn -> logicalColumn.name().equals(columnName))
+                  .collect(ImmutableList.toImmutableList());
+              if (matchingInputColumns.isEmpty()) {
+                throw new UnknownColumnException("SELECT", (ColumnReferenceExp) e);
+
+              }
+              if (matchingInputColumns.size() > 1) {
+                throw new KsqlException("Ambiguous column " + columnName + " in SELECT clause");
+              }
+
+              outputSchemaBuilder.add(Iterators.getOnlyElement(matchingInputColumns.iterator()));
```

Comment:
This is called `outputSchemaBuilder` but seems the `LogicalName` is from input. Why not use output alias as output logicalname if alias exists?

Reply:
Will add alias support only later.